source/mesh_deformation/free_surface.cc

```diff
@@ -307,15 +307,27 @@ namespace aspect
 
                 cell_vector = 0;
                 cell_matrix = 0;
+                Tensor<1,dim> direction;
+                if (advection_direction == SurfaceAdvection::geometric_normal)
+                  // Only for 2D
+                  {
```

Comment:
Add an assertion if it only works for a certain case.

Reply:
I'm sorry I've left these changes to `free_surface.cc` in. They were for testing whether a different normal would get rid of surface topography asymmetry, and they have nothing to do with this PR.